lib/data/locationConstraintParser.js

```diff
@@ -32,7 +33,35 @@ function parseLC() {
             clients[location].clientType = 'scality';
         }
         if (locationObj.type === 'aws_s3') {
-            // TODO
+            const s3Params = {
+                endpoint: `https://${locationObj.details.awsEndpoint}`,
+                // Non-file stream objects are not supported
+                // with SigV4 (node sdk)
+                // so note that we are using the default of
+                // signatureVersion v2
+                debug: false,
+                // Not implemented yet for streams in node sdk,
+                // and has no negative impact if stream, so let's
+                // leave it in for future use
+                computeChecksums: true,
+            };
+            // users can either include the desired profile name from their
+            // ~/.aws/credentials file or include the accessKeyId and
+            // secretAccessKey directly in the locationConfig
+            if (locationObj.details.credentialsProfile) {
+                s3Params.credentials = new AWS.SharedIniFileCredentials({
+                    profile: locationObj.details.credentialsProfile });
+            } else {
+                s3Params.accessKeyId =
+                    locationObj.details.credentials.accessKey;
+                s3Params.secretAccessKey =
+                    locationObj.details.credentials.secretKey;
+            }
+            clients[location] = new AWS.S3(s3Params);
```

Comment:
I don't want to block this PR for this, but it seems to me that `client[location]` should be an instance of a wrapper that embeds the actual AWS handle, instead of being the AWS handle, that would avoid "polluting" the multipleBackendGateway file with special cases for aws_s3.